### PR TITLE
Enable running cpu-mgr-multiNUMA e2e tests with Topology manager

### PR DESCRIPTION
**Pull request by Deepthidharwar**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

/kind cleanup

**What this PR does / why we need it**:
Currently, TopologyManager e2e tests have a copy of CPU Manager e2e test code and does not support MultiNUMA support. This PR addresses both these issues by code-refactor and re-use.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by Deepthidharwar** [comment]
@vpickard  @ConnorDoyle , @fromanirh , Please do review and if it looks ok, need ok-to-test.

**Comment by vpickard** [comment]
> /test pull-kubernetes-node-e2e-containerd

It looks like this test is failing consistently across the board. I have posted a message on #sig-testing to see the test env needs some tidying up. For reference, here is  a link to this job run history:

https://prow.k8s.io/job-history/kubernetes-jenkins/pr-logs/directory/pull-kubernetes-node-e2e-containerd

**Comment by Deepthidharwar** [comment]
Failing test is not cpu-manager's test but SR-IOV device plugin tests. Looking into this.

**Comment by fromanirh** [comment]
> Failing test is not cpu-manager's test but SR-IOV device plugin tests. Looking into this.

It's a bug I introduced in the tests which triggers when the tests is run on a box without any SRIOV device. The logic which tries to detect if we need to skip the tests for this very reason is flawed.

**Comment by Deepthidharwar** [comment]
@fromanirh , SRIOV detection logic is reporting error on systems that does not have one enabled https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/topology_manager_test.go#L172 . This should not error out.

**Comment by fromanirh** [comment]
> @fromanirh , SRIOV detection logic is reporting error on systems that does not have one enabled https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/topology_manager_test.go#L172 . This should not error out.

Correct. It should read like
```
if len(pciPerNuma) == 0 {
   return -1, false
}
```

**Comment by Deepthidharwar** [comment]
#88721 has the fix for this as discussed above.

**Comment by Deepthidharwar** [comment]
The changes made as part of this patch runs fine on systems with fix in #88721.

**Review by ffromani — COMMENTED** [review]
lgtm assuming the inline comment is handled

File: test/e2e_node/cpu_manager_test.go [hunk 1]
```diff
@@ -258,266 +258,328 @@ func enableCPUManagerInKubelet(f *framework.Framework, cleanStateFile bool) (old
...
+	err = f.PodClient().MatchContainerOutput(pod.Name, pod.Spec.Containers[0].Name, expAllowedCPUsListRegex)
+	framework.ExpectNoError(err, "expected log not found in container [%s] of pod [%s]",
+		pod.Spec.Containers[0].Name, pod.Name)
 
-		// Skip CPU Manager tests altogether if the CPU capacity < 2.
-		if cpuCap < 2 {
-			e2eskipper.Skipf("Skipping CPU Manager tests since the CPU capacity < 2")
-		}
+	ginkgo.By("by deleting the pods and waiting for container removal")
+	deletePods(f, []string{pod.Name})
```
[the remark this answers is not in the document]
Comment (Deepthidharwar) [reply test/e2e_node/cpu_manager_test.go]: Good catch.

**Comment by fromanirh** [comment]
lgtm now. Thanks for the updates!

**Comment by derekwaynecarr** [comment]
the refactoring was clean.  thanks for the follow-up.

/approve
/lgtm

**Pull request merged**